chain/epoch-manager/src/epoch_sync.rs

```diff
@@ -12,12 +13,102 @@ use near_primitives::epoch_sync::{
 use near_primitives::hash::CryptoHash;
 use near_primitives::types::validator_stake::ValidatorStake;
 use near_primitives::types::{AccountId, ApprovalStake, BlockHeightDelta, EpochId};
+use near_primitives::version::BLOCK_HEADER_V3_PROTOCOL_VERSION;
 use near_store::Store;
 use near_store::adapter::StoreAdapter;
+use near_store::adapter::chain_store::ChainStoreAdapter;
+use near_store::adapter::epoch_store::EpochStoreAdapter;
 use near_store::merkle_proof::MerkleProofAccess;
 use rayon::iter::{IntoParallelIterator, ParallelIterator};
 use tracing::instrument;
 
+/// Function to extend epoch sync proof by one epoch.
+/// - Retrieve the existing epoch sync proof from the store
+/// - Validate that the existing proof is consistent with the new epoch to be added, i.e.
+///   the proof exists upto the previous epoch.
+/// - Generate and append the new epoch data to the existing proof.
+/// - Updated proof back to the store.
+pub fn extend_epoch_sync_proof(
```

Comment:
Should this be split into two functions, one doing the extension and one doing the store read and write? That way the extension itself can be tested?